android/authentication.md

```diff
@@ -1,615 +1,704 @@
+[Amazon Cognito Federated Identities](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-identity.html) on the other hand, is a way to authorize your users to use AWS services. With an identity pool, you can obtain temporary AWS credentials with permissions you define to access other AWS services directly or to access resources through Amazon API Gateway.
 
-Create and configure an `AuthUIConfiguration` object and set its properties.
+When working together, Cognito User Pools acts as a source of user identities (identity provider) for the Cognito Federated Identities. Other sources can be OpenID, Facebook, Google, etc. AWS Amplify uses User Pools to store your user information and handle authorization, and it leverages Federated Identities to manage user access to AWS Resources, for example allowing a user to upload a file to an S3 bucket.
 
-* To present the Email and Password user `SignInUI`, set `userPools` to `true`.
+Ensure you have [installed and configured the Amplify CLI and library]({%if jekyll.environment == 'production'%}{{site.amplify.docs_baseurl}}{%endif%}/js/start).
+{: .callout .callout--info}
 
-* To present Facebook or Google  user `SignInUI`, add `signInButton(FacebookButton.class)` or `signInButton(GoogleButton.class)`.
+## How it works
 
-* To change the logo, use the `logoResId`.
+The AWSMobileClient manages your application session for authentication related tasks. The credentials it pulls in can be used by other AWS services when you call a `.default()` constructor. The Amplify category examples in this documentation use this by default, however [you can also use this with any AWS service via the generated SDK clients](manualsetup#direct-aws-service-access).
 
-* To change the background color, use `backgroundColor`.
+### State tracking
 
-* To cancel the sign-in flow, set `.canCancel(true)`.
+`AWSMobileClient` offers on-demand querying for the "login state" of a user in the application. For instance, you can check if the user is signed-in or not and present an appropriate screen. This is done through a couple of mechanisms:
 
-* To change the font in the sign-in views, use the `fontFamily` method and pass in the string that represents a font family.
+- `isLoggedIn` property defined as a BOOLEAN for the most simple use cases
+- `currentUserState` used for more advanced scenarios, such as determining if the user has Guest credentials, Authenticated with User Pools, has Federated credentials, or has signed out.
 
-* To draw the `backgroundColor` full screen, use `fullScreenBackgroundColor`.
+This allows you to write workflows in your application based on the state of the user and what you would like to present on different screens. The `AWSMobileClient` also offers realtime notifications on user state changes which you can register for in your application using `.addSignInStateListener` as in the code below.
```

Comment:
.addSignInStateListener => .addSignInStateListener()

Reply:
I think there are a couple more of those in the doc.

Reply:
Done